client/result.go

```diff
@@ -174,54 +205,82 @@ func (c *Client) getAllResultObjects(ctx context.Context, teamID, cloudID string
 	var scrollId string
 	var cur = 0
 	var totalItems = 0
+	var request *resultObjectRequest
 
 	for cur == 0 || cur < totalItems {
-		tmp, err := c.getResultObjects(ctx, scrollId, link.Href, buf)
-		if err != nil {
-			return res, err
+		if cur == 0 {
```

Comment:
Think about using a different variable for your initial condition to make it more readable.